lib/WasmReader/WasmByteCodeGenerator.cpp

```diff
@@ -1107,14 +1107,9 @@ WasmBytecodeGenerator::EmitReturnExpr(EmitInfo *lastStmtExprInfo)
         }
 
         m_writer.Conv(retOp, 0, retExprInfo.location);
-        if (!lastStmtExprInfo)
-        {
-            ReleaseLocation(&retExprInfo);
-        }
     }
     m_writer.AsmBr(m_funcInfo->GetExitLabel());
-
-    return EmitInfo();
+    return retExprInfo;
```

Comment:
Very good point. Somewhat misleading was the fact that the original example actually type checked under spec. I'll have to double check what is going on in spec. Closing.